pkg/3scale/amp/operator/upgrade.go

```diff
@@ -36,6 +36,9 @@ func NewUpgradeApiManager(b *reconcilers.BaseReconciler, apiManager *appsv1alpha
 }
 
 func (u *UpgradeApiManager) Upgrade() (reconcile.Result, error) {
+	// TODO Upgrade should be grouped by component and not by feature.
```

Comment:
I think this totally depends on what you are doing.
Sometimes it seems it's desirable to upgrade by component and sometimes it seems it is desirable to be done by feature, depending on whether you want to do a specific set of changes before another etc...